### PR TITLE
feat(senkora/fs)!: Implement fs apis 

### PR DESCRIPTION
**Pull request by TiranexDev**
- [x] *exists-Related*
    - [x] exists 
    - [x] existsFile
    - [x] existsDirectory
- [x] *delete-Related*
    - [x] deleteFile
    - [x] deleteDirectory
- [x] *create-Related*
    - [x] createDirectory

### CODE REVIEW

File: src/modules/fs/api.c [hunk 1]
```diff
@@ -20,22 +26,119 @@ int writeToFile(const char *filename, const char *data) {
...
     fclose(f);
     return str;
 }
+
+int deleteFile(const char *filename)
+{
+    char command[256];
+    sprintf(command, "rm %s", filename);
+
+    if (system(command) == 0)
```
Comment (Fire-The-Fox) [inline src/modules/fs/api.c:57]: Don't call system commands, what if they aren't present? see man page for `unlink(2)`

File: src/modules/fs/api.c [hunk 2]
```diff
@@ -20,22 +26,119 @@ int writeToFile(const char *filename, const char *data) {
...
+    else
+    {
+        return 0;
+    }
+}
+
+int deleteDirectory(const char *dirname)
+{
+    char command[256];
+    sprintf(command, "rm -r %s", dirname);
```
Comment (Fire-The-Fox) [inline src/modules/fs/api.c:70]: Don't call system commands, what if they aren't present?

File: src/modules/fs/api.c [hunk 3]
```diff
@@ -20,22 +26,119 @@ int writeToFile(const char *filename, const char *data) {
...
+        {
+            return 0; // Exists as a file
+        }
+    }
+}
+
+int createDirectory(const char *dirname)
+{
+    char command[256];
+    sprintf(command, "mkdir %s", dirname);
```
Comment (Fire-The-Fox) [inline src/modules/fs/api.c:134]: Don't call system commands, what if they aren't present? see man page for `mkdir(2)`

File: src/modules/fs/api.c [hunk 4]
```diff
@@ -20,22 +26,110 @@ int writeToFile(const char *filename, const char *data) {
...
+    }
+    else
+    {
+        return 0;
+    }
+}
+
+int deleteDirectory(const char *dirname)
+{
+    if (rmdir(dirname) == 0)
```
Comment (Fire-The-Fox) [inline src/modules/fs/api.c:66]: This function will fail on non-empty directory

**Comment by Fire-The-Fox** [comment]
Everything is fine except those commits, keep things professional otherwise no.

**Comment by TiranexDev** [comment]
> Everything is fine except those commits, keep things professional otherwise no.

Okay, sorry

**Pull request merged**